src/test/java/com/squareup/kotlinpoet/ClassNameTest.kt

```diff
@@ -137,12 +137,12 @@ class ClassNameTest {
     } catch (expected: IllegalArgumentException) {
     }
 
-    try {
-      ClassName.get(Array<Any>::class)
-      fail()
-    } catch (expected: IllegalArgumentException) {
-    }
-
+    // TODO
```

Comment:
See https://github.com/square/kotlinpoet/issues/86#issuecomment-305071978. This is the language leaking implementation details of the JVM 😢 .

I'm not sure what to do here yet...